File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

// prototypes
int get_words(string str);
int get_letters(string str);
int get_sentences(string str);

int main(void)
{
    // gets user input
    string user_input_string = get_string("Text: ");

    // calculates words, letters and sentences
    int words = get_words(user_input_string);
    int letters = get_letters(user_input_string);
    int sentences = get_sentences(user_input_string);

    // caluclates average letters per 100 words and average sentences per 100 words
    float average_letters = (float)letters / words * 100;
    float average_sentences = (float)sentences / words * 100;

    // calculates Coleman-Liau index
    float index = 0.0588 * average_letters - 0.296 * average_sentences - 15.8;

    // round index to get the grade
    int grade = (int)round(index);

    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", grade);
    }

    return 0;
}

int get_words(string str)
{
    int counter = 0;
    for (int i = 0; str[i] != '\0'; i++)
    {
        // if character is space count as a word
        if (str[i] == ' ')
        {
            counter++;
        }
    }
    // return counter +1 because of last word
    return ++counter;
}

int get_letters(string str)
{
    int counter = 0;
    for (int i = 0; str[i] != '\0'; i++)
    {
        // if character is uppercase or lowercase letter increment counter
        if ((str[i] > 64 && str[i] < 91) || (str[i] > 96 && str[i] < 123))
        {
            counter++;
        }
    }
    return counter;
}

int get_sentences(string str)
{
    int counter = 0;
    for (int i = 0; str[i] != '\0'; i++)
    {
        // if character is . or ! or ? count as end of sentence increment counter
        if (str[i] == '.' || str[i] == '!' || str[i] == '?')
        {
            counter++;
        }
    }
    return counter;
}

File: pset6/readability/readability.py
from cs50 import get_string


def main():
    # gets text from user
    input_string = get_string("Text: ")

    # get number of words, letters and sentences from input text
    words = get_words(input_string)
    letters = get_letters(input_string)
    sentences = get_sentences(input_string)
    
    # calculate average letters and average sentences per 100 in text
    average_letters = letters / words * 100
    average_sentences = sentences / words * 100

    # calculate index
    index = 0.0588 * average_letters - 0.296 * average_sentences - 15.8
    
    # get grade
    grade = int(round(index))

    # print grade
    if (grade >= 16):
        print("Grade 16+")
    elif (grade < 1):
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")


def get_words(string):
    counter = 0
    for char in string:
        # if char is space 
        if (char == " "):
            counter += 1
    counter += 1
    return counter


def get_letters(string):
    counter = 0
    for char in string:
        # if char is letter increase counter
        if (char.isalpha()):
            counter += 1
    return counter


def get_sentences(string):
    counter = 0
    end_signs = [".", "!", "?"]
    for char in string:
        if char in end_signs:
            counter += 1
    return counter


if __name__ == "__main__":
    main()